js/gated-content/src/components/MediaPlayer.vue

```diff
@@ -48,20 +50,43 @@ export default {
   },
   computed: {
     player() {
-      return this.media.field_media_source.startsWith('youtube') ? 'youtube' : 'vimeo';
+      return this.media.field_media_source.startsWith('youtube')
+        ? 'youtube'
+        : 'vimeo';
     },
     videoId() {
-      let embedObj = this.media.field_media_video_embed_field;
       if (this.media.field_media_source === 'youtube_playlist') {
-        embedObj = embedObj.match(/(\?|&)v=([^&#]+)/).pop();
-        return embedObj;
+        return this.mediaUrl.match(/(\?|&)v=([^&#]+)/).pop();
       }
       // If the video matches Vimeo's private link format, return the full url.
-      if (embedObj.match(/^https?:\/\/(www\.)?vimeo.com\/([0-9]*)(\/[a-zA-Z0-9]+)$/)) {
-        return embedObj;
+      if (
+        this.mediaUrl.match(
+          /^https?:\/\/(www\.)?vimeo.com\/([0-9]*)(\/[a-zA-Z0-9]+)$/,
+        )
+      ) {
+        return this.mediaUrl;
       }
       return this.media.field_media_video_id;
     },
+    mediaUrl() {
+      return this.media.field_media_video_embed_field;
+    },
+    timecode() {
+      const matches = this.mediaUrl.match(/[&?#]t=((\d+)h)?((\d+)m)?(\d+)s?/);
```

Comment:
The details about using such regular expression:

![image](https://user-images.githubusercontent.com/5138333/140305384-5708be1a-9cbf-45a6-a525-9a2217626331.png)

![image](https://user-images.githubusercontent.com/5138333/140305456-9ab888b4-bb8e-4ff8-a155-d51856ba7543.png)